examples/flock/index.js

```diff
@@ -0,0 +1,209 @@
+      await wait(100);
+      retry = false;
+      await upsertData(jsonObj, query);
+    } else {
+      errorCount += 1;
+      console.log(`ERROR Unable to commit.\n${err}\n`);
+    }
+  }
+}
+
+async function filterTweet(tweet) {
+  const userMentions = [];
+  const usersObject = [];
+  usersObject[tweet.user.id_str] = 'uid(u)';
+  tweet.entities.user_mentions.forEach((element, index) => {
+    let uid;
+    if (usersObject[element.id_str] != undefined) {
+      uid = usersObject[element.id_str];
+    } else {
+      uid = `uid(m${index+1})`;
+      usersObject[element.id_str] = uid;
+    }
+    userMentions.push({
+      'uid': uid,
+      'user_id': element.id_str,
+      'dgraph.type': 'User',
+      'user_name': element.name,
+      'screen_name': element.screen_name,
+    });
+  });
+  const hashtags = [];
+  tweet.entities.hashtags.forEach((element) => {
+    hashtags.push(element.text);
+  });
+  const author = {
+    'uid': `uid(u)`,
+    'user_id': tweet.user.id_str,
+    'dgraph.type': 'User',
+    'user_name': tweet.user.name,
+    'screen_name': tweet.user.screen_name,
+    'description': tweet.user.description,
+    'friends_count': tweet.user.friends_count,
+    'followers_count': tweet.user.followers_count,
+    'verified': tweet.user.verified,
+    'profile_banner_url': tweet.user.profile_banner_url,
+    'profile_image_url': tweet.user.profile_image_url,
+  };
+  const userObj = {
+    'uid': `uid(t)`,
+    'id_str': tweet.id_str,
+    'dgraph.type': 'Tweet',
+    'created_at': new Date(tweet.created_at),
+    'message': tweet.text,
+    'urls': tweet.urls,
+    'hashtags': hashtags,
+    'mention': userMentions,
+    'author': author,
+  };
+  return userObj;
+}
+
+async function buildQuery(tweet) {
+  const usersObject = [];
+  const query = [];
+
+  query.push(`t as var(func: eq(id_str, "${tweet.id_str}"))`);
+  query.push(`u as var(func: eq(user_id, "${tweet.author.user_id}"))`);
+
+  usersObject[tweet.author.user_id] = 'u';
+
+  tweet.mention.forEach((element, index) => {
+    let name;
+    if (usersObject[element.user_id] != undefined) {
+      name = usersObject[element.user_id];
+    } else {
+      name = `m${index+1}`;
+      query.push(`${name} as var(func: eq(user_id, ${element.user_id}))`);
+      usersObject[element.user_id] = name;
+    }
+  });
+
+  const finalQuery = `query {${query.join('\n')}}`;
+  return finalQuery;
+}
+
+function reportStats() {
+  const now = new Date().getTime();
+  // tslint:disable-next-line no-console
+  console.log(`STATS Tweets: ${totalTweets}, Failues: ${failureCount}, Retries: ${retryCount}, \
+Errors: ${errorCount}, Commit Rate: ${(totalTweets-oldTotalTweets)/(LOG_INTERVAL_TIME/1000)}  Total Time: ${now - startStatus} ms`);
+  oldTotalTweets = totalTweets;
+}
+
+async function wait(time) {
+  return new Promise((resolve) => {
+    const id = setTimeout(
+        () => {
+          clearTimeout(id);
+          resolve();
+        },
+        time,
+    );
+  });
+}
+
```

Comment:
Since this is test code `const wait = time => new Promise(resolve => setTimeout(resolve, time))` is good enough and shorter. `const id` isn't used anyway so no reason to keep it around or call `clearTimeout`